src/window.js

```diff
@@ -82,30 +69,14 @@ var ToolboxWindow = GObject.registerClass({
         });
     }
 
-    getUnlockState() {
-        return this.toolbox.getUnlockState();
-    }
-
-    setUnlockState(index, v) {
-        this.toolbox.setUnlockState(index, v);
-    }
-
-    unlock() {
-        const unlockState = this.getUnlockState();
-        const index = unlockState.findIndex(state => !state);
-        this.setUnlockState(index, true);
-        this.emit('unlock-state-changed');
-
-        // Update state of the common lockscreen in this window
-        this._lockscreen.locked = !this.getUnlockState()[0];
+    getMainLockscreen() {
+        return this._lockscreen;
```

Comment:
I find that it would be more consistent to have this be a simple getter, since that's what we do for the toolbox.